### ISSUE TITLE
Shader cycling (n/m) resetting to root shader folder

### ISSUE THREAD

**Issue by ferro4prez** [libretro/RetroArch#11050]
## Description

Using the n/m keys to cycle through shaders results in resetting to the root shader folder, essentially making this unusable and pointless.

### Expected behavior

Choose a shader (ie, from the CRT folder), use n/m to cycle through all the different shaders *within the crt folder*.  In short, the n/m keys should cycle within the folder of the last-selected shader preset.

### Actual behavior

Basically the equivalent of going to a clothing store and rather than looking at the clothes with our eyes, we're looking through a catalogue of descriptions of clothing, choosing one without being able to see it, and repeating that process requiring 10-12 clicks/movements every single time.

### Workaround (not good)

Setting the default shader folder to the subfolder you want to n/m through.

###

Thank you!

**Comment by johanbcn** [comment]
I did some work on that feature some time ago because of a related configuration change, and what I saw on the code is that it is intended to be used to cycle through **your shader presets** (as when you use the "Save Shader Preset As" option).

**Comment by ferro4prez** [comment]
> I did some work on that feature some time ago because of a related configuration change, and what I saw on the code is that it is intended to be used to cycle through **your shader presets** (as when you use the "Save Shader Preset As" option).

Ah I see...the fact that hundreds of shaders are included however should expect an easy way to navigate them.  Currently just to audition a different shader it's like 10 keypresses *at least*, seeing that you have to navigate the entire folder structure every single time, not to mention all the menu navigation.

**Comment by bslenul** [comment]
If you want to cycle bundled presets, as a workaround you can change `Settings > Directory > Shaders` to point to `shaders/shaders_slang/crt/` folder (or "shaders_glsl") for example to cycle CRT shaders.

**Comment by johanbcn** [comment]
Yeah, that's what I expected too when I first encountered this feature.

The issue resides on the fact that the list of available shaders is built everytime a core starts, and that relies on file system access performance, which is attrocious on some portable systems. That's the reason it's limited to user shader presets, which is bound to be small enough to avoid being a performance hit everytime a core starts.

In my opinion the only way to solve this would be to manage a prebuilt list of shaders, kind of like the game playlists.

**Comment by RexBorova** [comment]
All of this. Main reason I stay away from applying and using shaders and used video filters from settings> videos instead. Video filters doesn't even benefit my Shield TV due to CPU usage, but for the convenience for saving 15 seconds of time to cycle shaders, I'll make that sacrifice
